daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1851,17 +1852,24 @@ private[lf] object SBuiltin {
               machine.enterApplication(action, Array(SEValue(SToken)))
           }
         } else {
-          // TODO #15882: check with ledger that required authority change is allowed
-
-          // do scoped authority change; (TX) Node.Authority *IS* created
-          machine.ptx.beginGainAuthority(
-            required = required
-          ) match {
-            case ptx =>
-              machine.ptx = ptx
-              machine.pushKont(KCloseGainAuthority)
-              machine.enterApplication(action, Array(SEValue(SToken)))
-          }
+          Control.Question[Question.Update](
+            Question.Update.NeedAuthority(
+              holding = holding,
+              requesting = requesting,
+              callback = { () =>
+                // do scoped authority change; (TX) Node.Authority *IS* created
+                machine.ptx.beginGainAuthority(
+                  required = required
+                ) match {
+                  case ptx =>
+                    machine.ptx = ptx
+                    machine.pushKont(KCloseGainAuthority)
+                    val c = machine.enterApplication(action, Array(SEValue(SToken)))
+                    machine.setControl(c)
+                }
+              },
```

Comment:
what is the rational of using a pattern matching here (I know it is not from this PR).

Reply:
just consistency with other `ptx` state changers which return errors via `Either`, i.e `beginExercise`